src/main/java/com/linkedin/kmf/services/SignalFxMetricsReporterService.java

```diff
@@ -0,0 +1,197 @@
+  private static final Logger LOG = LoggerFactory.getLogger(SignalFxMetricsReporterService.class);
+
+  private final String _name;
+  private final List<String> _metricNames;
+  private final List<String> _dimensions;
+  private final int _reportIntervalSec;
+  private final ScheduledExecutorService _executor;
+  private final MetricRegistry _metricRegistry;
+  private final SignalFxReporter _signalfxReporter;
+  private final String _signalfxUrl;
+  private final String _signalfxToken;
+  
+  private MetricMetadata _metricMetadata;
+  private Map<String, SettableDoubleGauge> _metricMap; 
+  private Map<String, String> _dimensionsMap;
+  
+  public SignalFxMetricsReporterService(Map<String, Object> props, String name) throws Exception {
+      
+    SignalFxMetricsReporterServiceConfig config = new SignalFxMetricsReporterServiceConfig(props);
+    
+    _name = name;
+    _metricNames = config.getList(SignalFxMetricsReporterServiceConfig.REPORT_METRICS_CONFIG);
+    _dimensions = config.getList(SignalFxMetricsReporterServiceConfig.SIGNALFX_METRIC_DIMENSION);
+    _reportIntervalSec = config.getInt(SignalFxMetricsReporterServiceConfig.REPORT_INTERVAL_SEC_CONFIG);
+    _signalfxUrl = config.getString(SignalFxMetricsReporterServiceConfig.REPORT_SIGNALFX_URL);
+    _signalfxToken = config.getString(SignalFxMetricsReporterServiceConfig.SIGNALFX_TOKEN);
+    
+    if (_signalfxToken.length() < 1) {
+      throw new IllegalArgumentException("SignalFx token is not configured");
+    }
+    
+    _executor = Executors.newSingleThreadScheduledExecutor();
+    _metricRegistry = new MetricRegistry();
+    _metricMap = new HashMap<String, SettableDoubleGauge>();
+    _dimensionsMap = new HashMap<String, String>();
+    setUpDimensionMap();
+    
+    SignalFxReporter.Builder sfxReportBuilder = new SignalFxReporter.Builder(
+        _metricRegistry,
+        _signalfxToken
+    );
+    if (_signalfxUrl.length() > 1) {
+      SignalFxEndpoint signalFxEndpoint = getSignalFxEndpoint(_signalfxUrl);
+      _signalfxReporter = sfxReportBuilder.setEndpoint(signalFxEndpoint).build();
+    } else {
+      _signalfxReporter = sfxReportBuilder.build();
+    }
+    
+    _metricMetadata = _signalfxReporter.getMetricMetadata();    
+    _signalfxReporter.start(_reportIntervalSec, TimeUnit.SECONDS);    
+  }
+
+  @Override
+  public synchronized void start() {
+    _executor.scheduleAtFixedRate(new Runnable() {
+      @Override
+      public void run() {
+        try {
+          captureMetrics();          
+        } catch (Exception e) {
+          LOG.error(_name + "/SignalFxMetricsReporterService failed to report metrics", e);
+        }
+      }
+    }, _reportIntervalSec, _reportIntervalSec, TimeUnit.SECONDS);
+    LOG.info("{}/SignalFxMetricsReporterService started", _name);
+  }
+
+  @Override
+  public synchronized void stop() {
+    _executor.shutdown();
+    LOG.info("{}/SignalFxMetricsReporterService stopped", _name);
```

Comment:
You also need to stop the `_signalfxReporter` here, otherwise you can't guarantee that everything's been sent correctly before shutting down.